### ISSUE TITLE
Add support for local files in custom pipelines

### ISSUE THREAD

**Issue by vvvm23** [huggingface/diffusers#1141]
**Is your feature request related to a problem? Please describe.**
Currently there is no way to pass a local pipeline file to `DiffusionPipeline` as a custom pipeline. This makes the development process when adding custom pipelines quite difficult, as it requires pushing to `main` first or clearing the cache (see [this comment](https://github.com/huggingface/diffusers/issues/841#issuecomment-1292727322))

**Describe the solution you'd like**
When a local path is specified, a custom pipeline should be loaded using the local file.

**Describe alternatives you've considered**
There are workarounds as specified in the above link. Personally, I also just added a `__main__` block to the custom pipeline file and executed it directly - but this is less than ideal.

Thanks :hugs:

**Comment by patrickvonplaten** [comment]
Yes very much agree! I'll try to tackle it this week - in case anybody is interested I'm more than happy to help someone guide to merge a PR here.

**Comment by vvvm23** [comment]
Nice, I'll close once merged :)))
